lib/src/context.dart

```diff
@@ -140,8 +140,30 @@ abstract class ContextHelpers {
   // when used with [_jsifyNewContext].
   static dynamic unjsifyNewContext(dynamic interopContext) {
     if (interopContext != null && hasProperty(interopContext, _reactDartContextSymbol)) {
-      return getProperty(interopContext, _reactDartContextSymbol);
+      return _ContextValue.unwrapIfNeeded(getProperty(interopContext, _reactDartContextSymbol));
     }
     return interopContext;
   }
 }
+
+/// A wrapper around a value that can't be stored in its raw form
+/// within a JS object (e.g., a Dart function).
+class _ContextValue {
```

Comment:
Is this functionally any different than `_JsBackedMapValue`? Could just combine them into a common utility if they're not different

Reply:
You're right, I was keeping them separate because I wasn't sure if they should be private. But it probably just makes more sense to add it to the utilities file. Updated in 32b58a7